lib/validation.js

```diff
@@ -0,0 +1,56 @@
+class ValidationError extends Error {
+  constructor(msg) {
+    super(msg)
+  }
+}
+
+const validate = (data, schema) => {
+  const errors = [];
+  for (const key in schema) {
+    // Check that field is present if it is required
+    if (schema[key].required) {
+      if (!(key in data)) {
+        errors.push(new ValidationError(`Missing required field ${key}.`))
+        continue;
+      }
+    }
+
+    // Run field-specific validation function if present
+    if (key in data) {
+      if (schema[key].validate) {
+        try {
+          // validation functions should throw ValidationError
+          schema[key].validate(data[key])
+        } catch (e) {
+          if (e instanceof ValidationError) {
+            errors.push(e)
+          } else {
+            throw e
+          }
+        }
+      }
+    }
+  }
+
+  // Throw the errors if there are any
+  if (errors.length < 1) {
+    return true
+  } else {
+    throw { validationErrors: true, errors }
```

Comment:
How is catching these errors supposed to work? When I catch this object it will print `ValidationErrors:true, []`, so I don't know if the pushed objects are dying after leaving scope or if I'm just using it wrong.

Reply:
Yeah, this is a problem with having Express render the Error objects directly. Doesn't matter anymore since that's all being eliminated.